salt/modules/vagrant.py

```diff
@@ -0,0 +1,462 @@
+# -*- coding: utf-8 -*-
+'''
+Work with virtual machines managed by vagrant
+
+    .. versionadded:: Oxygen
+'''
+
+# Import python libs
+from __future__ import absolute_import, print_function
+import logging
+
+# Import salt libs
+import salt.cache
+import salt.utils
+import salt.utils.stringutils
+from salt.exceptions import CommandExecutionError, SaltCacheError, SaltInvocationError
+import salt.ext.six as six
+
+if six.PY3:
+    import ipaddress
+else:
+    import salt.ext.ipaddress as ipaddress
+
+log = logging.getLogger(__name__)
+
+__virtualname__ = 'vagrant'
+
+
+def __virtual__():
+    '''
+    run Vagrant commands if possible
+    '''
+    # noinspection PyUnresolvedReferences
+    if salt.utils.path.which('vagrant') is None:
+        return False, 'The vagrant module could not be loaded: vagrant command not found'
+    return __virtualname__
+
+
+def _update_cache(name, vm_):
+    vm_cache = salt.cache.Cache(__opts__, expire=13140000)  # keep data for ten years
```

Comment:
Why are we keeping the cache for 10 years?

This should be able to be configured, and it looks like elsewhere it defaults to `10`

Reply:
I am purposely misusing cache as a database. (Somebody suggested that I should do that in the saltify cloud driver.)  

The cache stores the mapping between the Salt node name and the Vagrant machine name (and directory). So when the user types "vagrant.destroy x1" I can determine which vagrant client machine she means. It needs to be maintained forever, and ten years seems like a nice approximation of that.

Reply:
The other option would be to use a real database.  We could use sqlite, but this felt like it fit the application better.

Reply:
This should use a real database then.  Sqlite is probably fine.

If you wanted it to be pluggable you can use `sdb` https://docs.saltstack.com/en/latest/topics/sdb/